templates/template-primary-secondary.js

```diff
@@ -0,0 +1,59 @@
+import '../components/colors/colors.js';
+import { css, html, LitElement } from 'lit-element/lit-element';
+
+class TemplatePrimarySecondary extends LitElement {
+
+	static get properties() {
+		return {};
+	}
+
+	static get styles() {
+		return css`
+			.container {

```

Comment:
Is the `<div>` container even needed, or could you apply these styles to the host itself using `:host`?

Reply:
I sometimes find myself on the fence about this.  If they are on the host, technically the consumer can override them, and in this case if they change the `display` the rest of the properties (and potentially others for nested elements) don't make sense and the layout breaks.  On the flip-side, sometimes giving the consumer the power is valid and opens up the API for them... but they ultimately take responsibility for it.

Reply:
Right, that's a really good point. Haha so which case does this fall under? The only trade-off I can think of is 1 extra element, which for something like buttons is a big deal but for a page-level template is not. So I'd be good keeping the extra `<div>` for that safety.

Reply:
Yeah I'm not sure it really matters in this case.  I think the pro for using `host` in this case is that it avoids a potentially unnecessary element, and the goal isn't to expose any aspect of the layout... is that true?  

`d2l-card` and `d2l-dialog` are two other good examples - they have a specific internal layout they are trying to provide, and if the consumer could override it, things are going to go haywire.  Card is especially interesting, because we expect the user might set `display` on it, in order to render with other cards in grid-like layout.  

I think `display` is one of those muddy properties that, depending on it's value, can affect both the layout behavior of the element in its parent context (ex. `inline-block`, `inline`, et.al.), and also the layout behavior of children within it (ex. `flex`, `grid`).  In some cases (like `d2l-card`), we need the extra element to separate these two concerns, but in other cases it's not so clear.

Anyway, not to belabor the topic.  Just wanted to air it as a point of non-clarity for many elements, at least for me... 🤔

Reply:
Totally, and since this is relying on `display: grid` I think it makes a lot of sense to hide that away in this container. So ignore my initial comment @wongvincent, keep the container. :)

Having said all that, you should probably set an initial display value for the host since by default it'll be an inline element which probably isn't what you want. And then as soon as you do that, you'll need to provide the `[hidden]` selector:

```css
:host { display: block; }
:host([hidden]) { display: none; }
```